### PR TITLE
[Laravel] Add RemoveDumpDataDeadCodeRector

### PR DESCRIPTION
**Pull request by hirenkeraliya**
This PR adds a new rule `RemoveDumpDataDeadCodeRector` to remove the unnecessary dump data code.

We are Humans And Human makes mistake sometimes we miss removing dump data code like `dd` or `dump` function calls.

That's why I added this new rector class to remove code that the developer forgot to remove.

### CODE REVIEW

File: src/Rector/FuncCall/RemoveDumpDataDeadCodeRector.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    /**
+     * @param FuncCall|StaticCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof FuncCall) {
+            return null;
+        }
+
+        if (! $this->isName($node->name, 'dd') && ! $this->isName($node->name, 'dump')) {
```
Comment (samsonasik) [inline src/Rector/FuncCall/RemoveDumpDataDeadCodeRector.php:79]: `isNames($node, ['dump', 'dd])` can be used

File: src/Rector/FuncCall/RemoveDumpDataDeadCodeRector.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class];
```
Comment (samsonasik) [inline src/Rector/FuncCall/RemoveDumpDataDeadCodeRector.php:67]: It should use Expression instead, so check against Expression expr is FuncCall direct remove FuncCall on If_ cond possibly cause error

Comment (samsonasik) [reply src/Rector/FuncCall/RemoveDumpDataDeadCodeRector.php:67]: Another approach is using FuncCall, but when remove, verify first that parent attribute is Expression stmt

File: tests/Rector/FuncCall/RemoveDumpDataDeadCodeRector/Fixture/skip_in_if_condition.php.inc [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Rector\Laravel\Tests\Rector\FuncCall\RemoveDumpDataDeadCodeRector\Fixture;
+
+class SkipWithArguments
```
Comment (samsonasik) [inline tests/Rector/FuncCall/RemoveDumpDataDeadCodeRector/Fixture/skip_in_if_condition.php.inc:5]: should be `SkipAsIfCondition`

File: src/Rector/FuncCall/RemoveDumpDataDeadCodeRector.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    /**
+     * @param FuncCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof FuncCall) {
+            return null;
+        }
```
Comment (samsonasik) [inline src/Rector/FuncCall/RemoveDumpDataDeadCodeRector.php:78]: can be removed, it always a `FuncCall` as defined in `getNodeTypes()`: 

```suggestion
```

**Comment by samsonasik** [comment]
You can fix CI by run:

```
vendor/bin/rector && composer fix-cs
```

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Could you update PR title to something like : `[Laravel] Add RemoveDumpDataDeadCodeRector`?

**Comment by hirenkeraliya** [comment]
> Could you update the PR title to something like: `[Laravel] Add RemoveDumpDataDeadCodeRector`?

Sorry.

**Comment by samsonasik** [comment]
Thank you @hirenkeraliya

**Pull request merged**